devito/data/allocators.py

```diff
@@ -306,6 +306,21 @@ def put_local(self):
         return self._node == 'local'
 
 
+class NumpyAllocator(MemoryAllocator):
```

Comment:
docstring? basically, to be used whenever the user wants to hand ownership of some data to devito, right?

Reply:
It is the opposite of handing ownership of data to Devito in my opinion? 

If I'm the user, and let's say I used some proprietary file reader that insists on allocating its own memory, I need to be able to define a Devito (Function) symbol that is backed by the data I already have loaded, without copying it to Devito's allocated memory. Does that make sense?

Reply:
yep, it does. Though note that without a `clear_cache` the data will stick around even if out of scope in user code, since it's now attached to the NumpyAllocator itself